components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Services/EndpointRegistry.cs

```diff
@@ -388,46 +388,48 @@ public sealed class EndpointRegistry : IEndpointRegistry, IApplicationEndpointRe
             var unchanged = 0;
             var removed = 0;
 
-            // Remove or disable an endpoint
-            foreach (var item in remove) {
-                try {
-                    // Only touch applications the discoverer owns.
-                    if (item.DiscovererId == discovererId) {
-                        if (hardDelete) {
-                            var device = await _iothub.GetAsync(item.DeviceId);
-                            // First we update any registration
-                            var existingEndpoint = device.ToEndpointRegistration(false);
-                            if (!string.IsNullOrEmpty(existingEndpoint.SupervisorId)) {
-                                await SetSupervisorTwinSecretAsync(existingEndpoint.SupervisorId,
-                                    device.Id, null);
+            if (!(result.RegisterOnly ?? false)) {
+                // Remove or disable an endpoint
+                foreach (var item in remove) {
+                    try {
+                        // Only touch applications the discoverer owns.
+                        if (item.DiscovererId == discovererId) {
+                            if (hardDelete) {
+                                var device = await _iothub.GetAsync(item.DeviceId);
+                                // First we update any registration
+                                var existingEndpoint = device.ToEndpointRegistration(false);
+                                if (!string.IsNullOrEmpty(existingEndpoint.SupervisorId)) {
+                                    await SetSupervisorTwinSecretAsync(existingEndpoint.SupervisorId,
+                                        device.Id, null);
+                                }
+                                // Then hard delete...
```

Comment:
Are we checking somehow if the endpoint is used in an active publishing job? In this case, we should not delete even when is inactive/disabled

Reply:
Yes, that is true.  But this would be a larger work item.  I will add this to our backlog.